controller/scheduler/fleet.py

```diff
@@ -337,16 +337,16 @@ def state(self, name):
             "deactivating": "down",
         }
         try:
-            # NOTE (bacongobbler): this call to ._get_unit() also acts as a pre-emptive check to
+            # NOTE (bacongobbler): this call to ._get_unit() acts as a pre-emptive check to
             # determine if the job no longer exists (will raise a RuntimeError on 404)
-            unit = self._get_unit(name)
+            self._get_unit(name)
```

Comment:
Correct. We have a check in the models [when the model is destroyed](https://github.com/deis/deis/blob/master/controller/api/models.py#L286-L288). Calling fleet.destroy() removes the unit from fleet. Instead of polling for 20 minutes to wait for the request to eventually time out polling for the unit state, `_get_unit()` raises a RuntimeError if the [request fails 3 times](https://github.com/deis/deis/blob/master/controller/scheduler/fleet.py#L58-L65). It's an optimization as noted in the comment :)